File: solaria-admin/src/types/menu.ts
export interface Menu {
  id: number;
  code: string;
  name: string;
  description: string;
  price: number;
  image_url: string;
  category: string;
  created_at: string;
  updated_at: string;
}

export type CreateMenuInput = Omit<Menu, "id" | "created_at" | "updated_at">;
export type UpdateMenuInput = Partial<CreateMenuInput>;

File: solaria-backend/models/menu.go
package models

import "time"

type Menu struct {
    ID          int       `json:"id"`
    Code        string    `json:"code"`
    Name        string    `json:"name"`
    Description string    `json:"description"`
    Price       float64   `json:"price"`
    ImageURL    string    `json:"image_url"`
    Category    string    `json:"category"`
    CreatedAt   time.Time `json:"created_at"`
    UpdatedAt   time.Time `json:"updated_at"`
}

type CreateMenuRequest struct {
    Code        string  `json:"code" validate:"required"`
    Name        string  `json:"name" validate:"required"`
    Description string  `json:"description"`
    Price       float64 `json:"price" validate:"required"`
    ImageURL    string  `json:"image_url"`
    Category    string  `json:"category" validate:"required"`
}

type UpdateMenuRequest struct {
    Name        string  `json:"name"`
    Description string  `json:"description"`
    Price       float64 `json:"price"`
    ImageURL    string  `json:"image_url"`
    Category    string  `json:"category"`
}